### PR TITLE
Add flake8 linter workflow

### PR DESCRIPTION
**Pull request by drikusroor**
### Background
I've recently opened a PR (#865) that will add automated unit tests using Github workflow for every PR and commit on the master branch. The suggestion was made to also add `flake8` linting, though in a separate PR, which you are looking at right now. I think it'd be a **good idea to merge that other PR first** and then append the workflow by merging this one.

### Changes
- Add `flake8` linting
- Add a `flake8` linting job to the workflow (continue after error is set to true as there were still quite some linting issues)
- Rename the `unit_tests.yml` to `ci.yml` as it now contains more jobs than just unit tests

### Documentation
I've added a small section to `README.md` that explains how to run the `flake8` linter.

### Test Plan
If the Python CI workflow succeeds the changes from this PR work as expected.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes ~with multiple different prompts.~ (prompts N/A)
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly ~and comprehensively.~ (needn't be comprehensive)
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,8 @@ ai_settings.yaml
 .vscode
 auto-gpt.json
 log.txt
+
+# Coverage reports
+.coverage
+coverage.xml
+htmlcov/
```
Comment (nponeccop) [inline .gitignore]: The final newline is missing. It's considered a whitespace error by git

File: main.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from scripts.main import main
```
Comment (nponeccop) [inline main.py]: Again the newline.

File: tests/context.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import sys
+import os
+
+sys.path.insert(0, os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '../scripts')))
```
Comment (nponeccop) [inline tests/context.py]: The newline

**Review by nponeccop — CHANGES_REQUESTED** [review]
I also suggest adding flake8 and the tests in a different PRs, and using the following small list of pep8 fixes:

python -m flake8 --select E303,W293,W291,W292,E305

This way we will start having pep8 but we won't break many PRs, and we gradually extend the list introducing minimal breakage.

**Review by nponeccop — DISMISSED** [review]

**Comment by drikusroor** [comment]
> I also suggest adding flake8 and the tests in a different PRs, and using the following small list of pep8 fixes:
> 
> python -m flake8 --select E303,W293,W291,W292,E305
> 
> This way we will start having pep8 but we won't break many PRs, and we gradually extend the list introducing minimal breakage.

The unit tests itself are added in a separate PR (here #865) already. I don't know if that's what you meant.

As for the `python -m flake8 --select E303,W293,W291,W292,E305` command: Are you suggesting to first use this command and then in later PRs we gradually be more strict about it, or am I understanding you wrong? :-)

**Comment by drikusroor** [comment]
> > I don't know if that's what you meant.
> 
> Well, I apologize then. Yes as for getting stricter you are right.

No worries, I was just wondering 😅

**Comment by richbeales** [comment]
Hi, your unit test PR has been merged, please can you resolve the conflicts in this one? Thanks

**Review by nponeccop — APPROVED** [review]

**Review by richbeales — APPROVED** [review]

**Pull request merged**